portal-impl/src/com/liferay/portal/template/TemplatePortletPreferences.java

```diff
@@ -34,29 +33,38 @@
 	public String getPreferences(Map<String, Object> preferences)
 		throws ReadOnlyException {
 
-		PortletPreferencesImpl portletPreferencesImpl =
-			new PortletPreferencesImpl();
+		StringBundler sb = new StringBundler();
+
+		sb.append("<portlet-preferences>");
 
 		for (Map.Entry<String, Object> entry : preferences.entrySet()) {
-			Object value = entry.getValue();
+			sb.append("<preference><name>");
+			sb.append(entry.getKey());
+			sb.append("</name>");
+
+			String[] values = _NULL_VALUE;
```

Comment:
In the old logic, we were only taking in String and String[] values, we were throwing away all other types, including nulls. How come here we start to add them in as "NULL_VALUE"?